swarm/network/stream/stream.go

```diff
@@ -666,18 +666,17 @@ func peerStreamIntervalsKey(p *Peer, s Stream) string {
 	return p.ID().String() + s.String()
 }
 
-func (c client) AddInterval(start, end uint64) (err error) {
-	i := &intervals.Intervals{}
-	err = c.intervalsStore.Get(c.intervalsKey, i)
-	if err != nil {
+func (c *client) AddInterval(start, end uint64) (err error) {
+	i := new(intervals.Intervals)
+	if err = c.intervalsStore.Get(c.intervalsKey, i); err != nil {
 		return err
 	}
 	i.Add(start, end)
 	return c.intervalsStore.Put(c.intervalsKey, i)
 }
 
-func (c client) NextInterval() (start, end uint64, err error) {
-	i := &intervals.Intervals{}
+func (c *client) NextInterval() (start, end uint64, err error) {
+	i := new(intervals.Intervals)
```

Comment:
Nitpick, but this is not the style that most Go programs use. Generally you create a new pointer value with `&smth{}` and not `new(smth)`. I am pretty sure this is much more common across our codebase as well.

Reply:
solved